.NET/Microsoft.Recognizers.Text.DataDrivenTests/TestHelpers.cs

```diff
@@ -0,0 +1,313 @@
+            }
+            return null;
+        }
+
+        private static IModel GetCustomModelFor(string language)
+        {
+            switch (language)
+            {
+                case Culture.Chinese:
+                    return new NumberModel(
+                        AgnosticNumberParserFactory.GetParser(AgnosticNumberParserType.Number, new ChineseNumberParserConfiguration()),
+                        new NumberExtractor(ChineseNumberMode.ExtractAll));
+            }
+            return null;
+        }
+    }
+
+    public static class TestModelExtensions
+    {
+        public static bool IsNotSupported(this TestModel testSpec)
+        {
+            return testSpec.NotSupported.HasFlag(Platform.dotNet);
+        }
+
+        public static bool IsNotSupportedByDesign(this TestModel testSpec)
+        {
+            return testSpec.NotSupportedByDesign.HasFlag(Platform.dotNet);
+        }
+
+        public static DateObject GetReferenceDateTime(this TestModel testSpec)
+        {
+            object dateTimeObject;
+            if (testSpec.Context.TryGetValue("ReferenceDateTime", out dateTimeObject))
+            {
+                return (DateObject)dateTimeObject;
+            }
+            return DateObject.Now;
+        }
+    }
+
+    public static class TestUtils
+    {
+        public static string GetCulture(string source)
+        {
+            var langStr = source.Substring(source.LastIndexOf('_') + 1);
+            switch (langStr)
+            {
+                case "Chs":
+                    return Culture.Chinese;
+                case "Spa":
+                    return Culture.Spanish;
+                case "Eng":
+                    return Culture.English;
+                case "Por":
+                    return Culture.Portuguese;
+                case "Fra":
+                    return Culture.French;
+                default:
+                    return Culture.English;
+            }
+        }
+
+        public static bool EvaluateSpec(TestModel spec, out string message)
+        {
+            if (string.IsNullOrEmpty(spec.Input))
+            {
+                message = $"spec not found";
+                return true;
+            }
+            if (spec.IsNotSupported())
+            {
+                message = $"input '{spec.Input}' not supported";
+                return true;
+            }
+            if (spec.IsNotSupportedByDesign())
+            {
+                message = $"input '{spec.Input}' not supported by design";
+                return true;
+            }
+            message = string.Empty;
+            return false;
+        }
+
+        public static string GetModelName(string source)
+        {
+            return source;
+        }
+
+        public static string GetExtractorName(string source)
+        {
+            return source.Replace("Base", "").Replace("Extractor", "").Replace("Parser", "");
```

Comment:
Strings as constants to facilitate maintenance later.

Reply:
Replaced all .Replace in several places by calling a single method (SanitizeSourceName:  https://github.com/southworkscom/Recognizers-Text/blob/e12d3f040fce71cb29d336b2f0703deec5c7d8b8/.NET/Microsoft.Recognizers.Text.DataDrivenTests/TestHelpers.cs#L337)